src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_NUMBER = 5;
+
+    private int index;
+    private Figure randomFigure;
+    private Random random = new Random();
+    private Figure defaultFigure = new Circle();
```

Comment:
When you create defaultFigure as new Circle(), default figure will have random radius, but then you change it to default one, maybe it will be better to add some logic which will prevent creation of random circle when it is not required, for example you can add to consteructor boolean parameter isDefault.